packages/cli/src/cmds/assert/strategy/strategy.ts

```diff
@@ -0,0 +1,34 @@
+// @ts-ignore
+import { Event } from '@appland/models';
+import Assertion from '../assertion';
+import { AppMapData } from '../../../appland/types';
+import { Scope } from '../types';
+
+export default abstract class Strategy {
+  protected abstract scope: Scope;
+  protected abstract isEventApplicable(event: Event): Boolean;
```

Comment:
Just a suggestion, but if `Strategy` and `Scope` are strongly connected, in that you're always adding a new `Scope` string when creating a new `Strategy`, this can be simplified by the type checker:

```ts
interface Strategy {
  isEventApplicable(event: Event): boolean;
}

const Strategies = {
  sql_query: {
    isEventApplicable(event: Event) {
      return Boolean(event.sql);
    },
  },
  http_server_request: {
    isEventApplicable(event: Event) {
      return Boolean(event.httpServerRequest);
    },
  },
} as const;

type Scope = keyof typeof Strategies;

export default Strategies;
```

Of course, this requires all of the the `Strategy` implementations to exist in a single object. YMMV. Something to play around with.

Reply:
Thanks for the idea. I think we may want to have a strategy that does not iterate over events, but appmaps only at some point. So I'd keep it as is to be more flexible if needed.